compiler/src/test/java/com/squareup/anvil/compiler/internal/reference/TopLevelPropertyReferenceTest.kt

```diff
@@ -0,0 +1,189 @@
+package com.squareup.anvil.compiler.internal.reference
+
+import com.google.common.truth.Truth.assertThat
+import com.squareup.anvil.compiler.api.AnvilContext
+import com.squareup.anvil.compiler.api.CodeGenerator
+import com.squareup.anvil.compiler.api.GeneratedFile
+import com.squareup.anvil.compiler.compile
+import com.squareup.anvil.compiler.internal.reference.Visibility.INTERNAL
+import com.squareup.anvil.compiler.internal.reference.Visibility.PRIVATE
+import com.squareup.anvil.compiler.internal.reference.Visibility.PUBLIC
+import com.tschuchort.compiletesting.KotlinCompilation.ExitCode.OK
+import org.jetbrains.kotlin.descriptors.ModuleDescriptor
+import org.jetbrains.kotlin.descriptors.PropertyDescriptor
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.resolve.scopes.DescriptorKindFilter
+import org.jetbrains.kotlin.resolve.source.getPsi
+import org.junit.Test
+import java.io.File
+
+class TopLevelPropertyReferenceTest {
+
+  @Test fun `top level properties are parsed for PSI and Descriptor APIs correctly`() {
```

Comment:
same here for having tests with more explicit expectations / use-cases

Reply:
Done